app/src/demoAbstract/java/com/jasperapps/happyfriend/common/di/CommonDi.kt

```diff
@@ -102,9 +108,22 @@ object CommonDi {
     fun provideSharedPreferences(@ApplicationContext context: Context): SharedPreferences {
         return PreferenceManager.getDefaultSharedPreferences(context)
     }
-
     @Provides
-    fun provideAnalytics(): Analytics {
+    fun provideToLogAnalytics(): ToLogAnalytics {
         return ToLogAnalytics()
     }
+
+    @Provides
+    fun provideFirebaseAnalytics(@ApplicationContext context: Context): TheirFirebaseAnalytics {
+        return TheirFirebaseAnalytics.getInstance(context)
+    }
+
+    @Provides
+    fun provideAnalytics(
+        toLogAnalytics: ToLogAnalytics,
+        theirFirebaseAnalytics: TheirFirebaseAnalytics,
+    ): Analytics {
+        val ourAnalytics = OurFirebaseAnalytics(theirFirebaseAnalytics)
+        return CompoundAnalytics(ourAnalytics, toLogAnalytics)
+    }
```

Comment:
most probably you mixed it up when resolving merge conflicts, but we need to use `ToLogAnalytics` analytics for demo (demoAbstract) build type. We dont what to have firebase analytics in demo apps.